iroha/src/isi.rs

```diff
@@ -78,12 +78,11 @@ impl Execute for RegisterBox {
                 }
                 _ => Err("Unsupported instruction.".to_string()),
             },
-            IdBox::PeerId(peer_id) => match self.object {
-                IdentifiableBox::Domain(domain) => Register::<Peer, Domain>::new(*domain, peer_id)
+            IdBox::WorldId => match self.object {
+                IdentifiableBox::Domain(domain) => Register::<World, Domain>::new(*domain, WorldId)
+                    .execute(authority, world_state_view),
+                IdentifiableBox::Peer(peer) => Register::<World, Peer>::new(*peer, WorldId)
```

Comment:
I'm not sure why Peer is here either. Generally a peer can be a validating or an observing peer. But the validation or observation is done with respect to an Account that is associated with a Peer. So all permissions are on the Peer's Account and not necessarily the peer itself.

Reply:
Yes, you are right. Though the exact staking mechanism is not yet implemented and Peer keys are not yet associated with account. I think this needs to be done in a separate PR as a separate task.

Right now we simply `Register` a new peer and as soon as the block with this ISI is committed, the peer starts participating in consensus.